cypress/integration/patch/configure-patch.ts

```diff
@@ -0,0 +1,182 @@
+/// <reference types="Cypress" />
+/// <reference path="../../support/index.d.ts" />
+
+type ProjectVariants = Array<{
+  name: string;
+  displayName: string;
+  tasks: string[];
+}>;
+interface PatchProject {
+  tasks?: string[];
+  variants?: ProjectVariants;
+}
+type VariantsTasks = Array<{
+  name: string;
+  tasks: string[];
+}>;
+interface ConfigurePatchData {
+  id: string;
+  description: string;
+  author: string;
+  version: string;
+  status: string;
+  activated: boolean;
+  time: {
+    submittedAt: string;
+  };
+  project: PatchProject;
+  variantsTasks: VariantsTasks;
+}
+interface ConfigurePatchQuery {
+  data: {
+    patch: ConfigurePatchData;
+  };
+}
+
+describe("Configure Patch Page", () => {
+  let patch: ConfigurePatchData;
+  const unactivatedPatchId = "5e6bb9e23066155a993e0f1a";
+  before(() => {
+    cy.login();
+    cy.visit(`/patch/${unactivatedPatchId}`);
+    cy.listenGQL();
+    cy.waitForGQL("ConfigurePatch").then(({ responseBody }) => {
+      const { data } = responseBody as ConfigurePatchQuery;
+      patch = data.patch;
+    });
+  });
+  describe("Initial state reflects patch data", () => {
+    it("Configure patch path is present in url", () => {
+      cy.location().should((loc) =>
+        expect(loc.pathname).to.eq(
+          `/patch/${unactivatedPatchId}/configure/tasks`
+        )
+      );
+    });
+    it("Patch name input field value is patch description", () => {
+      cy.get("[data-cy=configurePatch-nameInput]")
+        .invoke("val")
+        .then((text) => {
+          expect(text).to.equal(patch.description);
+        });
+    });
+    it("First build variant in list is selected by default", () => {
+      const firstVariantDisplayNameInList =
+        patch.project.variants[0].displayName;
+      cy.get("[data-cy-selected=true]")
+        .invoke("text")
+        .should("eq", firstVariantDisplayNameInList);
+    });
+    it("Patch tasks are selected by default", () => {
+      const { variantsTasks } = patch;
+      variantsTasks.forEach(({ name, tasks }) => {
+        cy.get(`[data-cy-name=${name}]`)
+          .click()
+          .then(() => {
+            tasks.forEach((task) => {
+              cy.get(`[data-cy=configurePatch-${task}]`).should("be.checked");
+            });
+          });
+      });
+    });
+  });
+  describe("Configuring a patch", () => {
+    it("Can update patch description by typing into `Patch Name` input field", () => {
```

Comment:
This tests that we can type into the description input but it doesn't test that the description of the patch is actually updated. I think that will possible once the schedule button is hooked up.

Reply:
i can add the test now